pkg/machinery/config/configloader/configloader.go

```diff
@@ -34,7 +34,7 @@ func newConfig(source []byte) (config config.Provider, err error) {
 	// a special way.
 	for _, manifest := range manifests {
 		if talosconfig, ok := manifest.(*v1alpha1.Config); ok {
-			return talosconfig, nil
+			return v1alpha1.WrapReadonly(talosconfig, source), nil
```

Comment:
this is the gist of the PR :rofl: 

when machine config is read from file (or from network), we make it read-only and we preserve the `[]byte` source